feat: add abilty to use button as A-tag button with href

REJack opened the pull request.

PH7-Jack commented:
@REJack Thanks for your contribution. 
I'm making a code review

PH7-Jack commented on resources/views/components/button.blade.php:
```diff
@@ -1,6 +1,7 @@
-<button {{ $attributes->merge([
+@if(isset($href))
```
Don't need to use an if to switch between `a` and `button` tags, just replace the HTML tag
`<{{ $tag }} ..... > .... </{{ $tag }}>`

So, we can use
`<x-button tag="a" href="google.com" .... />`

PH7-Jack replied:
Add a tag option

PH7-Jack commented on src/View/Components/BaseButton.php:
```diff
@@ -106,6 +110,7 @@ protected function mergeData(array $data): array
         $attributes       = $data['attributes'];
         $data['disabled'] = (bool) $attributes->get('disabled');
         $data['classes']  = $this->getClasses();
+        $data['href']     = $this->href;
```
Why did you add it? Don't need to pass the class properties to `$data`
```suggestion
```

PH7-Jack commented on tests/Browser/Button/view.blade.php:
```diff
@@ -45,4 +45,13 @@
     <x-button outline info label="info" right-icon="home" />
     <x-button outline dark label="dark" right-icon="home" />
+
+    <x-button href="#" label="Label" icon="home" />
+    <x-button href="#" primary label="primary" icon="home" />
+    <x-button href="#" secondary label="secondary" icon="home" />
+    <x-button href="#" disabled positive label="positive" icon="home" />
+    <x-button href="#" disabled negative label="negative" icon="home" />
+    <x-button href="#" warning label="warning" icon="home" />
+    <x-button href="#" info label="info" icon="home" />
+    <x-button href="#" dark label="dark" icon="home" />
```
Missing a test to assert if the href can is rendered

PH7-Jack reviewed (changes requested):
Thanks for your contribution

REJack commented:
I've added a correct test and changed the approach, but without the "tag"-attribute IMHO it's odd and not really needed since you need a `href` for `a`-tags and they will not used on `button`-tags.

bernhardh commented:
Would be great, if this could be merged

PH7-Jack reviewed (approved):
Thanks for this contribution. Sorry for my long time to review
